### DISCUSSION TITLE
Proof-of-Trust: A Wealth-Unbiased Consensus Mechanism for Distributed Systems

### DISCUSSION THREAD

**Discussion by sanity** [freenet/freenet-core#799] [Blog]
#### Introduction

Traditional approaches to rate-limiting the creation of things like coins, tokens, and identities in distributed networks rely on computational (proof-of-work) or financial barriers (proof-of-stake) as a source of scarcity. While effective in some contexts, these methods are wasteful and unfairly favor those with more resources.

#### What is Proof-of-Trust?

Proof-of-Trust offers an alternative by utilizing the scarcity of reciprocal trust between individuals as the primary resource for rate-limiting various network activities. Unlike existing models, Proof-of-Trust does not inherently favor participants with greater computational power or financial means.

#### The Prisoner's Dilemma Analogy

In the classic Prisoner's Dilemma, two participants must choose between cooperating for mutual benefit or defecting for individual gain at the other's expense. In a similar vein, Proof-of-Trust involves two existing network participants who must decide to cooperate or defect when attempting to create a new user.

Participants must first stake something of value—such as their ability to create new users for a set period. The outcome varies based on their choices:

1. **Both Cooperate**: A new user is created.
2. **One Defects**: The defector gains two new users under their control, while the other participant is penalized by losing their staking privilege for a period.
3. **Both Defect**: Both participants are penalized.

#### Flexibility of Outcomes

The exact nature of the rewards and punishments is flexible and subject to further refinement. This allows for the fine-tuning of incentives to achieve desired operational outcomes, such as controlling the rate of new user creation.

#### Versatility in Application

While the creation of new users serves as an illustrative example, Proof-of-Trust can be applied to a wide range of network activities requiring some form of rate-limiting or conditional access.

#### The Ethical and Practical Benefits

Proof-of-Trust levels the playing field by not favoring those with more resources. In a world where social and economic status often determines access and influence, this mechanism offers a more equitable way to participate in a network.

Additionally, Proof-of-Trust naturally encourages meaningful human interaction. Because the system relies on trust between participants, people are motivated to get to know each other better. This need for trust not only enhances network security but also serves as a catalyst for building community and forming meaningful human connections.

#### Legal Considerations

Speculatively, joining the network could require users to consent to a legal agreement, collected by their sponsors. Analogous to the [GNU General Purpose License](https://en.wikipedia.org/wiki/GNU_General_Public_License), this contract would outline essential ethical norms and explicitly prohibits "cheating," including any form of coercive behavior. Non-compliance would risk penalties or exclusion, reinforcing the network's foundation of mutual trust.

**Comment by sulivanShu** [comment]
One thing to consider is the publicity given to the exchange. The risk of cheating is drastically reduced if the exchange is public, or semi-public, that is to say public within an interest group.

When it comes to contracts, people will take them as seriously as Facebook's "Terms of Service" or "Community Guidelines." There is no "Community" on the Internet, there are only relationships of interests and power which are organized into all kinds of political systems, most of them being dictatorships.

This is why I think that a program like Freenet must instead offer tools allowing agents to establish trust between themselves, without imposing a particular form of trust or designating a particular regulatory agent.

From a more general point of view, I'm not sure that Freenet's target audience is very receptive to "communities" governed by contracts, like those established by companies under the pretext of benevolence but which in reality embody their dictatorship and their interests.

**Comment by DamianHayden** [comment]
When building a new system, we must consider how much it would cost to exploit it.
Especially in the early stages, the network is so small that it is essentially very easy to compromise.
Even with that in mind, if incentives work more strongly and abuse can be minimized, that would be an excellent proof.

**Comment by bitmage** [comment]
Excuse my verbosity.  A number of ideas come to mind.

> "One Defects: The defector gains two new users under their control, while the other participant is penalized by losing their staking privilege for a period."

Doesn't this incentivize defecting?

> "Participants must first stake something of value"

How is that different from proof of stake?  If the thing you're staking is 1 to 1 with an account, then it basically amounts to voting.  I guess the mechanism itself prevents mass creation of accounts.  But would it also arbitrarily, unnecessarily slow down the growth and adoption of the network?

### Consent Based Communication

When I look at SMS right now, the platform inherently has no concept of consent.  So they are trying to layer a regulatory apparatus over the top, and it's creating insane hoops for marketers to jump through (A2P registration).  It can't be fun to be a marketer and deal with that.

FreeNet could:

- provide for consent based communication by default, so unwanted communication is simply dropped
- allow for delegation of "introducer" status to my contacts (aka [Web of Trust](https://en.wikipedia.org/wiki/Web_of_trust))
- allow ephemeral burner inboxes to be created for public postings, e.g. I'm selling my bike, or commenting on the president's wardrobe.  This can still be associated with my identity, but not confer permission to contact me directly or indefinitely.
- allow creation of permanent public inboxes (e.g. business contact)
- allow businesses to post their offerings publicly, and a marketplace of AI-driven personal aggregation tools to be built that would find products, friends, work opportunities relevant for the person's needs.  Invert the marketing model to put the individual in power.

Business owners want access to a relevant market.  They don't have any need or inherent desire to pay a bunch of middle men.  If you could create a system that more directly gets them what they want at lower cost and complexity, while also eliminating spam for individuals, everyone wins.  If you build a platform that people want to do business on, then you win.

### Trust Aggregation

Let's say you want to know whether someone is a spammer.  You could say "you need 3 people to vouch for you before you can send messages."  But then then a group of spammers could band together, vouch for each other, and start sending messages.

Well, what do we know about trust in the real world?  Trust is subjective, not absolute.  Spammers may earn the trust of spammers because they have a similar set of values.  But they may have a more difficult time gaining the trust of engineers, or farmers, or church-goers.  If you could begin to develop trust aggregates, you would likely see that there is a pattern of mistrust between many groups and the spammer group.  If the network is able to respond to that and block the unwanted communication, then have you accomplished the goal?

### Rate Limiting

What else would you be looking for in consensual communication?  An occasional acknowledgement?  Lack of "block" action?

What may be normal communication volume for a business may be enormous compared to the normal volume for an individual.  So I think that "appropriate flow rates" is very subjective and violations have to be signaled in some way by the offended parties.

One way you could handle this is have sane default rate limits for individuals, but allow people to apply for business accounts with higher transactional volumes (which could also be a source of revenues).  The businesses would have to describe their use case, sign an agreement that they will not abuse the privilege, etc.  And this would also give you a form of revenue.
